### PR TITLE
ARROW-7863: [C++][Python][CI] Ensure running HDFS related tests

### PR DESCRIPTION
**Pull request by kou**
libhdfs3 backend may be broken. It causes SEGV. So libhdfs3 backend
test is disabled for now.

### CODE REVIEW

**Comment by kou** [comment]
@ursabot crossbow submit test-conda-python-3.7-hdfs-2.9.2

File: .env [hunk 1]
```diff
@@ -35,6 +35,7 @@ PANDAS=latest
 DASK=latest
 TURBODBC=latest
 HDFS=2.9.2
+LIBHDFS3=2.3
```
Comment (wesm) [inline .env]: libhdfs3 2.3 is not supported at all. Version 2.2.31 is the last one we support but I'm not sure if that still works with the current set of conda-forge dependencies / compilers. We might propose reverting the feedstock to the last "good" version that uses the libhdfs C ABI

https://github.com/conda-forge/libhdfs3-feedstock/commit/97fd9320efd5cbac0145db60c9b867582e9a748a#diff-ae18fe4e01e61a9862ceaf165003213a

**Review by wesm — COMMENTED** [review]
cc @xhochy

**Comment by wesm** [comment]
Also to be honest I would be OK with removing support for libhdfs3 altogether since it is abandonware

File: ci/scripts/integration_hdfs.sh [hunk 2]
```diff
@@ -22,20 +22,57 @@ set -e
...
+
+function use_hadoop_home() {
+  unset ARROW_LIBHDFS_DIR
+  export HADOOP_HOME=$hadoop_home
+}
+
+function use_libhdfs_dir() {
+  unset HADOOP_HOME
+  export ARROW_LIBHDFS_DIR=$libhdfs_dir
+}
```
Comment (pitrou) [inline ci/scripts/integration_hdfs.sh]: @kou Could you explain the difference between the two? (use_hadoop_home vs. use_libhdfs_dir)
Are there two sources for libhdfs?

Comment (kou) [reply ci/scripts/integration_hdfs.sh]: We need to find `${HADOOP_HOME}/lib/native/libhdfs.so` when user uses `HADOOP_HOME`.
We need to find `${ARROW_LIBHDFS_DIR}/libhdfs.so` when user uses `ARROW_LIBHDFS_DIR`.

If `HADOOP_HOME` is defined, `ARROW_LIBHDFS_DIR` isn't used. So we need to test them separately.

**Comment by kou** [comment]
OK. I'll remove libhdfs3 support.

**Comment by kszucs** [comment]
We used to test libhdfs3 support until conda forge provided the right version, but in the last iteration I've removed it.

**Comment by kou** [comment]
@ursabot crossbow submit test-conda-python-3.7-hdfs-2.9.2

**Comment by kou** [comment]
@ursabot crossbow submit test-conda-python-3.7-hdfs-2.9.2

**Comment by kou** [comment]
@ursabot crossbow submit test-conda-python-3.7-hdfs-2.9.2

**Comment by kou** [comment]
Done.

**Comment by kou** [comment]
+1

File: .env (hunk 1 above)
Comment (xhochy) [reply .env]: I can build that for you when you provide me with a source tarball at a semi-official location. The original one seems to be vanished.

**Pull request closed without merging**